pages/pipelines/defining_steps.md.erb

```diff
@@ -71,20 +73,40 @@ steps:
       queue: 'deploy'
 ```
 
-See the [Buildkite Agent pipelines documentation](/docs/agent/cli-pipeline) for a full list of step types and supported options.
+## Step types
 
-## Dynamic pipelines
+There are four different step types you can use in your Buildkite pipelines:
 
-You can generate pipelines dynamically from your source code thanks to the fact that pipelines are uploaded by running a command on one of your Buildkite agents. This provides you with a large amount of flexiblity to structure your pipelines as you wish.
+* [Command Step](/docs/pipelines/command-step)
+* [Wait Step](/docs/pipelines/wait-step)
+* [Block Step](/docs/pipelines/block-step)
+* [Trigger Step](/docs/pipelines/trigger-step)
 
-For example you could generate a list of parallel test steps based upon the `test/*` directories within your repository. To do this you'd have a single step in your pipeline settings:
+## Customizing the pipeline file path
 
-```yml
-steps:
-  - command: ./scripts/buildkite_test_pipeline | buildkite-agent pipeline upload
+By default the pipeline upload step reads your pipeline definition from `.buildkite/pipeline.yml` in your repository. You can specify a different file path by adding it as the first argument:
+
+<%= image "custom-pipeline-file.png", width: 1174/2, height: 452/2, alt: "Screenshot of a custom pipeline file upload" %>
+
+A common use for custom file paths is when separating test and deployment steps into two completely separate pipelines, both using the same repository URL. For example, your test pipeline’s upload command could be:
+
+```
+buildkite-agent pipeline upload .buildkite/pipeline.yml
+```
+
+And your deployment pipeline’s upload command could be:
+
+```
+buildkite-agent pipeline upload .buildkite/pipeline.deploy.yml
 ```
 
-The `scripts/buildkite_test_pipeline` would be an executable file checked into your source repository, and could look something like the following:
+For a list of all command line options, see the [buildkite-agent pipeline upload](/docs/agent/cli-pipeline#uploading-pipelines) documentation.
+
+## Dynamic pipelines
+
+Because the pipeline upload step runs on your agent machine, you can generate pipelines dynamically using scripts from your source code. This provides you with a large amount of flexiblity to structure your pipelines as you wish.
```

Comment:
I think we can skip the 'a large amount of' bit, flexibility kind of speaks for itself.